components/script/dom/promise.rs

```diff
@@ -80,12 +80,13 @@ impl Drop for Promise {
 }
 
 impl Promise {
-    pub fn new(global: &GlobalScope, comp: &InCompartment) -> Rc<Promise> {
-        Promise::new_in_current_compartment(global, comp)
+    pub fn new(global: &GlobalScope) -> Rc<Promise> {
+        let comp = AlreadyInCompartment::assert(&global);
```

Comment:
We want to always enter a compartment, not panic if we have not entered one. We need to use `let compartment = JSAutoCompartment::new(cx, global.reflector().get_jsobject().get());` here.